core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncoders.java

```diff
@@ -76,20 +88,27 @@ private static HttpEncodingType determineEncoding(String acceptEncoding) {
             }
             if (encoding.contains("*")) {
                 starQ = q;
+            } else if (encoding.contains("br") && q > brQ) {
+                brQ = q;
             } else if (encoding.contains("gzip") && q > gzipQ) {
                 gzipQ = q;
             } else if (encoding.contains("deflate") && q > deflateQ) {
                 deflateQ = q;
             }
         }
-        if (gzipQ > 0.0f || deflateQ > 0.0f) {
-            if (gzipQ >= deflateQ) {
+        if (brQ > 0.0f || gzipQ > 0.0f || deflateQ > 0.0f) {
+            if (brQ != -1.0f && brQ >= gzipQ) {
+                return Brotli.isAvailable() ? HttpEncodingType.BR : null;
```

Comment:
Shouldn't we choose gzip or deflate if Brotli is not available?

Reply:
I think that's more reasonable. Thank you for comment!